config_dialog.py

```diff
@@ -113,15 +118,21 @@ def populate_triggers(self):
 
         self.triggers = list_triggers(conn)
 
+        if self.broken_triggers == None:
+            self.broken_triggers = sql_generator.list_invalid_triggers(conn, self.triggers)
+
         self._update_triggers_model()
 
     def delete_not_valid_triggers(self):
         if not self.triggers: return
 
+        conn = self.get_connection()
         invalid = []
         for trigger_id, src, trg in self.triggers:
             if not src or not trg:
                 invalid.append((trigger_id, src, trg))
+            elif (trigger_id, src, trg) in self.broken_triggers:
+                invalid.append((trigger_id, src, trg))
```

Comment:
Would be useful to have a comment why there are two different rules/tests for invalid triggers